cmd/neofs-cli/modules/object/util.go

```diff
@@ -284,3 +309,141 @@ func finalizeSession(cmd *cobra.Command, dst SessionPrm, tok *session.Object, ke
+	case err == nil:
+		common.PrintVerbose("Raw header received - object is singular.")
+		return nil
+	case errors.As(err, &errSplit):
+		common.PrintVerbose("Split information received - object is virtual.")
+	}
+
+	splitInfo := errSplit.SplitInfo()
+
+	// collect split chain by the descending ease of operations (ease is evaluated heuristically).
+	// If any approach fails, we don't try the next since we assume that it will fail too.
+
+	if idLinking, ok := splitInfo.Link(); ok {
+		common.PrintVerbose("Collecting split members using linking object %s...", idLinking)
+
+		addrObj.SetObject(idLinking)
+		prmHead.SetAddress(addrObj)
+		prmHead.SetRawFlag(false)
+		// client is already set
+
+		res, err := internal.HeadObject(prmHead)
+		common.ExitOnErr(cmd, "failed to get linking object's header: %w", err)
+
+		children := res.Header().Children()
+
+		common.PrintVerbose("Received split members from the linking object: %v", children)
+
+		// include linking object
+		return append(children, idLinking)
+	}
+
+	if idSplit := splitInfo.SplitID(); idSplit != nil {
+		common.PrintVerbose("Collecting split members by split ID...")
+
+		var query object.SearchFilters
+		query.AddSplitIDFilter(object.MatchStringEqual, idSplit)
+
+		var prm internal.SearchObjectsPrm
+		prm.SetContainerID(cnr)
+		prm.SetClient(cli)
+		prm.SetFilters(query)
+
+		res, err := internal.SearchObjects(prm)
+		common.ExitOnErr(cmd, "failed to search objects by split ID: %w", err)
+
+		members := res.IDList()
+
+		common.PrintVerbose("Found objects by split ID: %v", res.IDList())
+
+		return members
+	}
+
+	idMember, ok := splitInfo.LastPart()
+	if !ok {
+		common.ExitOnErr(cmd, "", errors.New("missing any data in received object split information"))
+	}
+
+	common.PrintVerbose("Traverse the object split chain in reverse...", idMember)
+
+	var res *internal.HeadObjectRes
+	chain := []oid.ID{idMember}
+	chainSet := map[oid.ID]struct{}{idMember: {}}
+
+	prmHead.SetRawFlag(false)
+	// split members are almost definitely singular, but don't get hung up on it
+
+	for {
+		common.PrintVerbose("Reading previous element of the split chain member %s...", idMember)
+
+		addrObj.SetObject(idMember)
+
+		res, err = internal.HeadObject(prmHead)
+		common.ExitOnErr(cmd, "failed to read split chain member's header: %w", err)
+
+		idMember, ok = res.Header().PreviousID()
+		if !ok {
+			common.PrintVerbose("Chain ended.")
+			break
+		}
+
+		if _, ok = chainSet[idMember]; ok {
+			common.ExitOnErr(cmd, "", fmt.Errorf("duplicated member in the split chain %s", idMember))
+		}
+
+		chain = append(chain, idMember)
+		chainSet[idMember] = struct{}{}
+	}
+
+	common.PrintVerbose("Looking for a linking object...")
+
+	var query object.SearchFilters
+	query.AddParentIDFilter(object.MatchStringEqual, obj)
+
+	var prmSearch internal.SearchObjectsPrm
+	prmSearch.SetClient(cli)
+	prmSearch.SetContainerID(cnr)
+	prmSearch.SetFilters(query)
+
+	resSearch, err := internal.SearchObjects(prmSearch)
```

Comment:
So time needed for a `DELETE` operation now depends on the object size?

Reply:
Yes, it is.